app/src/main/java/com/aerogear/androidshowcase/navigation/PushRegistrationFailedDialogFragment.java

```diff
@@ -0,0 +1,63 @@
+package com.aerogear.androidshowcase.navigation;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.support.v4.app.DialogFragment;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+
+import com.aerogear.androidshowcase.R;
+
+public class PushRegistrationFailedDialogFragment extends DialogFragment {
```

Comment:
We can probably refactor this and the service not available dialog fragment to make a single fragment that just has a different header and body and callback, wdyt?

Reply:
@aidenkeating I'm fine with recording a JIRA for making refactoring this, servicenotavailable, and warningdialog into a single class/factory instead of having it as part of this PR.

Reply:
@secondsun Due to time constraints, and the fact my focus is split at the moment, I think creating a separate Jira to address this makes sense